test/vips_test.rb

```diff
@@ -0,0 +1,131 @@
+require "test_helper"
+require "image_processing/vips"
+
+describe ImageProcessing::Vips do
+  include ImageProcessing::Vips
+
+  def assert_similar(expected, actual)
+    return if RUBY_ENGINE == "jruby"
+
+    a = Phashion::Image.new(expected.path)
+    b = Phashion::Image.new(actual.path)
+
+    distance = a.distance_from(b).abs
+
+    assert_operator distance, :<=, 2
+  end
+
+  def assert_dimensions(dimensions, file)
+    assert_equal dimensions, Vips::Image.new_from_file(file.path).size
+  end
+
+  def assert_type(type, file)
+    assert_equal type, File.extname(file.path)
+  end
```

Comment:
It's not enough to look only at file extension, as the extension doesn't have to match the image type. For example, I could just `mv some_image.jpeg some_image.png`, and the image will remain a JPEG image but will have a `.png` extension.

If you won't be able to figure out how to determine the image type using `ruby-vips`, you can use [MimeMagic](https://github.com/minad/mimemagic).